Code/GraphMol/SubstructLibrary/SubstructLibrary.cpp

```diff
@@ -69,6 +67,22 @@ struct Bits {
     }
   }
 
+  Bits(const FPHolderBase *fps, const TautomerQuery &m, bool recursionPossible,
+       bool useChirality, bool useQueryQueryMatches)
+      : fps(fps),
+        recursionPossible(recursionPossible),
+        useChirality(useChirality),
+        useQueryQueryMatches(useQueryQueryMatches) {
+    if (fps) {
+      const TautomerPatternHolder *tp =
+          dynamic_cast<const TautomerPatternHolder *>(fps);
+      PRECONDITION(tp, "TautomerQueries only support TautomerPatternHolders");
```

Comment:
Not sure if this should be a PRECONDITION (equivalent to an assertion; something which indicates a serious programming error) or maybe just throw a `ValueErrorException` (it's a mistake, but one which can be recovered from)

Reply:
I just made it a warning.